test-suite-serde-java/src/test/java/io/micronaut/problem/OutOfStockTest.java

```diff
@@ -0,0 +1,107 @@
+import io.micronaut.http.client.BlockingHttpClient;
+import io.micronaut.http.client.HttpClient;
+import io.micronaut.http.client.annotation.Client;
+import io.micronaut.http.client.exceptions.HttpClientResponseException;
+import io.micronaut.http.uri.UriBuilder;
+import io.micronaut.serde.ObjectMapper;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@MicronautTest
+public class OutOfStockTest {
+    @Inject
+    @Client("/")
+    HttpClient httpClient;
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @Test
+    void problemSerializedWithSerde() throws IOException {
+        //given:
+        BlockingHttpClient client = httpClient.toBlocking();
+
+        //when:
+        HttpClientResponseException e = assertThrows(HttpClientResponseException.class, () ->
+                client.exchange(HttpRequest.GET(UriBuilder.of("/product").build()))
+        );
+
+        //then:
+        assertEquals(HttpStatus.BAD_REQUEST, e.getStatus());
+        assertTrue(e.getResponse().getContentType().isPresent());
+        assertEquals("application/problem+json", e.getResponse().getContentType().get().toString());
+    }
+
+    @Disabled
+    @Test
+    void ProblemIsRenderedToMapWithSerde() throws IOException {
+        //given:
+        BlockingHttpClient client = httpClient.toBlocking();
+        //when:
+        Argument<?> okArg = Argument.of(String.class);
+        Argument<?> errorArg = Argument.of(Map.class);
+        HttpClientResponseException e = assertThrows(HttpClientResponseException.class, () ->
+                client.exchange(HttpRequest.GET(UriBuilder.of("/product").build()), okArg, errorArg)
+        );
+
+        //then:
+        assertEquals(HttpStatus.BAD_REQUEST, e.getStatus());
+        assertTrue(e.getResponse().getContentType().isPresent());
+        assertEquals("application/problem+json", e.getResponse().getContentType().get().toString());
+
+        //when:
+        Optional<Map> bodyOptional = e.getResponse().getBody(Map.class);
+
+        //then:
+        assertNotNull(bodyOptional);
+        assertTrue(bodyOptional.isPresent());
+
+        //and:
+        assertEquals(5, bodyOptional.get().keySet().size());
+        assertEquals(400, bodyOptional.get().get("status"));
+        assertEquals("Out of Stock", bodyOptional.get().get("title"));
+        assertEquals("Item B00027Y5QG is no longer available", bodyOptional.get().get("detail"));
+        assertEquals("https://example.org/out-of-stock", bodyOptional.get().get("type"));
+        assertEquals(Collections.singletonMap("product", "B00027Y5QG"), bodyOptional.get().get(("parameters")));
+    }
+
+    @Test
+    void ProblemIsRenderedToStringWithSerde() throws IOException {
```

Comment:
this test fails because the rendered json is: 

`"status":{"statusCode":400,"reasonPhrase":"Bad Request"}` instead of `status: 400`. I attempted to add `@Serdeable` annotation to `io.micronaut.problem.HttpStatusType` but it did not work. Any idea `@graemerocher`?

Reply:
`HttpStatusType` is not included in the `@SerdeImport`

Reply:
`HttpStatusType` is a class of `micronaut-problem` not an external class. I thought `@Serdeable` was the annotation to use.

Reply:
I added `@SerdeImport(HttpStatusType.class)`  but it does not help.

Reply:
I have created an issue https://github.com/micronaut-projects/micronaut-serialization/issues/109